src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1941,6 +1963,43 @@ def _dft_seminormal(self, mult='l2r'):
         snb = self.seminormal_basis(mult=mult)
         return matrix([vector(b) for b in snb]).inverse().transpose()
 
+    def _dft_modular(self):
+        r"""
+        Return the discrete Fourier transform when the characteristic divides the order of the group.
+
+        EXAMPLES::
+
+            sage: GF2S3 = SymmetricGroupAlgebra(GF(2), 3)
+            sage: GF2S3._dft_modular()
+            [1 0 0 0 1 0]
+            [0 1 0 0 0 1]
+            [0 0 1 0 0 1]
+            [0 0 0 1 1 0]
+            [1 0 0 1 1 0]
+            [0 1 1 0 0 1]
+
+        Check that :issue:`37751` is fixed::
```

Comment:
```suggestion
```
Well, not quite, but I can be precise about this with this suggestion. (There is no need to (effective) duplicate this doctest. It's something to try and avoid in practice.)